FTBFS on Debian unstable

## Problem

Fail to build from source.

## How to reproduce it

```
$ autoreconf -i
$ ./configure 
$ LANG=C make
```

Environments:

* Debian unstable
* g++ 7.3.0
```
$ g++ -v
Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/7/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 7.3.0-24' --with-bugurl=file:///usr/share/doc/gcc-7/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-7 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 7.3.0 (Debian 7.3.0-24) 

```

## What is expected?

No FTBFS.

## What is happened?

undefined reference occurs.

```
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../src/include/kytea -I./../include -DPKGDATADIR=\"/usr/local/share/kytea\" -g -Wall -O3 -MT kytea-struct.lo -MD -MP -MF .deps/kytea-struct.Tpo -c kytea-struct.cpp  -fPIC -DPIC -o .libs/kytea-struct.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../src/include/kytea -I./../include -DPKGDATADIR=\"/usr/local/share/kytea\" -g -Wall -O3 -MT kytea-struct.lo -MD -MP -MF .deps/kytea-struct.Tpo -c kytea-struct.cpp -o kytea-struct.o >/dev/null 2>&1
mv -f .deps/kytea-struct.Tpo .deps/kytea-struct.Plo
/bin/bash ../../libtool  --tag=CXX   --mode=link g++  -g -Wall -O3 -version-info 0:0:0  -o libkytea.la -rpath /usr/local/lib kytea.lo general-io.lo corpus-io-prob.lo corpus-io-eda.lo corpus-io-full.lo corpus-io-part.lo corpus-io-tokenized.lo corpus-io-raw.lo corpus-io.lo model-io.lo string-util.lo kytea-model.lo kytea-config.lo kytea-lm.lo feature-io.lo dictionary.lo feature-lookup.lo kytea-util.lo kytea-string.lo kytea-struct.lo liblinear/liblinear.la 
libtool: link: g++  -fPIC -DPIC -shared -nostdlib /usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/7/crtbeginS.o  .libs/kytea.o .libs/general-io.o .libs/corpus-io-prob.o .libs/corpus-io-eda.o .libs/corpus-io-full.o .libs/corpus-io-part.o .libs/corpus-io-tokenized.o .libs/corpus-io-raw.o .libs/corpus-io.o .libs/model-io.o .libs/string-util.o .libs/kytea-model.o .libs/kytea-config.o .libs/kytea-lm.o .libs/feature-io.o .libs/dictionary.o .libs/feature-lookup.o .libs/kytea-util.o .libs/kytea-string.o .libs/kytea-struct.o  -Wl,--whole-archive liblinear/.libs/liblinear.a -Wl,--no-whole-archive  -L/usr/lib/gcc/x86_64-linux-gnu/7 -L/usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/7/../../.. -lstdc++ -lm -lc -lgcc_s /usr/lib/gcc/x86_64-linux-gnu/7/crtendS.o /usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu/crtn.o  -g -O3   -Wl,-soname -Wl,libkytea.so.0 -o .libs/libkytea.so.0.0.0
libtool: link: (cd ".libs" && rm -f "libkytea.so.0" && ln -s "libkytea.so.0.0.0" "libkytea.so.0")
libtool: link: (cd ".libs" && rm -f "libkytea.so" && ln -s "libkytea.so.0.0.0" "libkytea.so")
libtool: link: (cd .libs/libkytea.lax/liblinear.a && ar x "/home/kenhys/work/debian/kytea/kytea/src/lib/liblinear/.libs/liblinear.a")
libtool: link: ar cru .libs/libkytea.a  kytea.o general-io.o corpus-io-prob.o corpus-io-eda.o corpus-io-full.o corpus-io-part.o corpus-io-tokenized.o corpus-io-raw.o corpus-io.o model-io.o string-util.o kytea-model.o kytea-config.o kytea-lm.o feature-io.o dictionary.o feature-lookup.o kytea-util.o kytea-string.o kytea-struct.o  .libs/libkytea.lax/liblinear.a/daxpy.o .libs/libkytea.lax/liblinear.a/ddot.o .libs/libkytea.lax/liblinear.a/dnrm2.o .libs/libkytea.lax/liblinear.a/dscal.o .libs/libkytea.lax/liblinear.a/linear.o .libs/libkytea.lax/liblinear.a/tron.o 
ar: `u' modifier ignored since `D' is the default (see `U')
libtool: link: ranlib .libs/libkytea.a
libtool: link: rm -fr .libs/libkytea.lax
libtool: link: ( cd ".libs" && rm -f "libkytea.la" && ln -s "../libkytea.la" "libkytea.la" )
make[3]: Leaving directory '/home/kenhys/work/debian/kytea/kytea/src/lib'
make[2]: Leaving directory '/home/kenhys/work/debian/kytea/kytea/src/lib'
Making all in bin
make[2]: Entering directory '/home/kenhys/work/debian/kytea/kytea/src/bin'
g++ -DHAVE_CONFIG_H -I. -I../../src/include/kytea  -I./../include -DPKGDATADIR='"/usr/local/share/kytea"'   -g -Wall -O3 -MT run-kytea.o -MD -MP -MF .deps/run-kytea.Tpo -c -o run-kytea.o run-kytea.cpp
mv -f .deps/run-kytea.Tpo .deps/run-kytea.Po
/bin/bash ../../libtool  --tag=CXX   --mode=link g++  -g -Wall -O3   -o kytea run-kytea.o  ../lib/libkytea.la 
libtool: link: g++ -g -Wall -O3 -o .libs/kytea run-kytea.o  ../lib/.libs/libkytea.so
/usr/bin/ld: ../lib/.libs/libkytea.so: undefined reference to `void kytea::checkValueVecEqual<int>(std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&)'
/usr/bin/ld: ../lib/.libs/libkytea.so: undefined reference to `void kytea::checkValueVecEqual<unsigned int>(std::vector<unsigned int, std::allocator<unsigned int> > const&, std::vector<unsigned int, std::allocator<unsigned int> > const&)'
/usr/bin/ld: ../lib/.libs/libkytea.so: undefined reference to `void kytea::checkValueVecEqual<short>(std::vector<short, std::allocator<short> > const&, std::vector<short, std::allocator<short> > const&)'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:396: kytea] Error 1
````

--- a/src/lib/kytea-util.cpp
+++ b/src/lib/kytea-util.cpp
@@ -53,6 +53,12 @@ template void checkValueVecEqual(const std::vector<vector<KyteaString> > * a, co
 template void checkValueVecEqual(const std::vector<int> * a, const std::vector<int> * b);
 template void checkValueVecEqual(const std::vector<KyteaString> * a, const std::vector<KyteaString> * b);
 
+template void checkValueVecEqual(const std::vector<unsigned int> & a, const std::vector<unsigned int> & b);
+template void checkValueVecEqual(const std::vector<short> & a, const std::vector<short> & b);
+template void checkValueVecEqual(const std::vector<vector<KyteaString> > & a, const std::vector<vector<KyteaString> > & b);
+template void checkValueVecEqual(const std::vector<int> & a, const std::vector<int> & b);
+template void checkValueVecEqual(const std::vector<KyteaString> & a, const std::vector<KyteaString> & b);
+
 // Vector equality checking function
 template <class T>
 void checkPointerVecEqual(const std::vector<T*> & a, const std::vector<T*> & b) {

